Removed BibTeX file type and included HTML and TXT types

Simone-Thai opened the pull request:
<!-- 
Describe the changes you have made here: what, why, ... 
Link issues that are fixed, e.g. "Fixes #333".
If you fixed a koppor issue, link it, e.g. "Fixes https://github.com/koppor/jabref/issues/47".
The title of the PR must not reference an issue, because GitHub does not support autolinking there.
-->
Fixes [koppor/#547](https://github.com/koppor/jabref/issues/547?fbclid=IwAR2TtODoezs2F35XsIifHrrVOXfKlSTWcAQACgjDp5JJQGbfcJXjnZxSBEU) by removing the BibTeX dropdown option. I have also added HTML and TXT filtering options as they are commonly used standard files.    

<img width="615" alt="Screen Shot 2022-10-23 at 11 34 49 pm" src="https://user-images.githubusercontent.com/110802893/197392448-17a354d9-d17c-4d1e-a570-66d2e0cefc98.png">

<!-- 
- Go through the list below. Please don't remove any items.
- [x] done; [ ] not done / not applicable
-->

- [x] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [x] Screenshots added in PR description (for UI changes)
- [x] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

Gpax971 commented:
Looks good, not sure about .txt files, but no harm in including it.
You may wish to adjust the imports to avoid changing them for no real reason. If you're using the seamless input thing, go to Settings > Editor > Code Style > Java, scroll down to the bottom and change import layout to
```
import javax.*
import java.*
blank line
import all other imports
blank line
import static all other imports
```

Racheal7 commented:
Hi Simone, the dropdown list looks good to me. However, I am not sure if only changing the dropdown list will solve the issue. 

Once the user selects the HTML or txt files and clicks "import", the software will check the file type and display messages on whether the file is imported successfully. However, currently, it only checks if the file is PDF or BibTex. I think if you changed the file types, you probably need to modify the `ImportHandler.java` as well, but I am not sure if it is within the scope of this issue. So you might need to discuss this with the maintainers.

Simone-Thai commented:
Thanks for the advice!
I've removed the the TXT filter option and put back the imports. 
Regarding checking the file types I've added a method `isHTMLFile` Boolean function but I'm not sure if this is needed right now. I've noticed that there is no importer for HTML files or other relevant methods to import them. I'll double check with the maintainers about this. 
I've also just edited small typos in some comments.

Simone-Thai commented:
This issue was opened on the wrong repository and has been reopened on the main repository. 
It can be found under [#9318](https://github.com/JabRef/jabref/pull/9318)
